build_kernel.sh

```diff
@@ -16,13 +17,15 @@ if git submodule status --cached agnos-kernel-sdm845/ | grep "^-"; then
 fi
 cd agnos-kernel-sdm845
 
-$DIR/tools/extract_tools.sh
+if [ "$ARCH" != "arm64" ] && [ "$ARCH" != "aarch64" ]; then
+  $DIR/tools/extract_tools.sh
 
-# Build parameters
-export ARCH=arm64
-export CROSS_COMPILE=$TOOLS/aarch64-linux-gnu-gcc/bin/aarch64-linux-gnu-
-export CC=$TOOLS/aarch64-linux-gnu-gcc/bin/aarch64-linux-gnu-gcc
-export LD=$TOOLS/aarch64-linux-gnu-gcc/bin/aarch64-linux-gnu-ld.bfd
+  # Build parameters
+  export ARCH=arm64
```

Comment:
ARCH should still be set since it's read by the kernel makefiles

Reply:
when ARCH is not set, it will build on the architecture of the system doing the build. on anything else (except arm64/aarch64), ARCH will be exported